### ISSUE TITLE
AttributeError: 'FlagEmbedding' object has no attribute 'embed_documents'

### ISSUE THREAD

**Issue by Cyboghostginx** [qdrant/fastembed#63]
```
from langchain.text_splitter import CharacterTextSplitter
from langchain.document_loaders import TextLoader

path_f = 'test.txt'

# Load and process the text
loader = TextLoader(path_f)
documents = loader.load()

#text_splitter = CharacterTextSplitter(chunk_size=300, chunk_overlap=70)
text_splitter = CharacterTextSplitter(chunk_size=1000, chunk_overlap=0)
texts = text_splitter.split_documents(documents)

with open(path_f, 'r') as file:
    content = file.read()
```

Hello I usually load my txt file for embedding with langchain textloader so as to split it into chunks. So I was able to do the embedding with code below but unable to add it to a vector store with chromadb

```
fe_model = Embedding(model_name="BAAI/bge-base-en-v1.5", max_length=512)

embeddings = fe_model.passage_embed(texts)
list(embeddings)
```

Code to add to chromadb 

```
from langchain.vectorstores import Chroma

vectorstore = Chroma.from_documents(documents=texts, embedding=Embedding(), persist_directory="./db2")
print("[+]Done")
```

Error - AttributeError: 'FlagEmbedding' object has no attribute 'embed_documents'
What could be wrong, I usually save my embedding this way to be able to use it with langchain llm like llama for free. Thanks!

**Comment by Anush008** [comment]
Hey. It seems you are using the `FlagEmbedding` class from the library to use with Langchain.
Whereas you should be importing

```python
from langchain.embeddings.fastembed import FastEmbedEmbeddings
```

**Comment by Cyboghostginx** [comment]
ModuleNotFoundError: No module named 'langchain.embeddings.fastembed'

But I'm getting this, I have langchain and fastembed installed

**Comment by Anush008** [comment]
Can you confirm your Langchain version?

**Comment by Cyboghostginx** [comment]
```
Name: langchain
Version: 0.0.331
Summary: Building applications with LLMs through composability
Home-page: https://github.com/langchain-ai/langchain
Author: 
Author-email: 
License: MIT
Location: /opt/homebrew/Caskroom/miniforge/base/lib/python3.10/site-packages
Requires: aiohttp, anyio, async-timeout, dataclasses-json, jsonpatch, langsmith, numpy, pydantic, PyYAML, requests, SQLAlchemy, tenacity
Required-by: 
```

**Comment by Anush008** [comment]
You are on an older version of Langchain and a version with the latest fastembed integration is not out on PyPi yet.

You can wait for a new release or do the following to get the latest version of Langchain from the source with the FastEmbed integration.

```bash
pip uninstall langchain
```
```bash
pip install git+https://git@github.com/langchain-ai/langchain.git@8d6faf56657070137a16669bc2420a08a5ab7f24#subdirectory=libs/langchain
```

**Comment by Cyboghostginx** [comment]
```
from langchain.embeddings.fastembed import FastEmbedEmbeddings

fe_model = FastEmbedEmbeddings(model_name="BAAI/bge-base-en-v1.5", max_length=512)

embeddings = fe_model.embed_documents(texts)
print(embeddings)
```
TypeError: TextEncodeInput must be Union[TextInputSequence, Tuple[InputSequence, InputSequence]]

**Comment by Anush008** [comment]
Could you recheck how `texts` is defined?

Here's a reference.

https://github.com/langchain-ai/langchain/blob/master/libs/langchain/tests/integration_tests/embeddings/test_fastembed.py#L16-L24

**Comment by Cyboghostginx** [comment]
```
from langchain.text_splitter import CharacterTextSplitter
from langchain.document_loaders import TextLoader

path_f = 'test.txt'

# Load and process the text
loader = TextLoader(path_f)
documents = loader.load()

#text_splitter = CharacterTextSplitter(chunk_size=300, chunk_overlap=70)
text_splitter = CharacterTextSplitter(chunk_size=1000, chunk_overlap=0)
texts = text_splitter.split_documents(documents)
```

This is how texts was defined. It is chunks of document

**Comment by Anush008** [comment]
You don't need to call `embed_documents()` explicitly with the docs. It's internal.

**Comment by Cyboghostginx** [comment]
```
from langchain.embeddings.fastembed import FastEmbedEmbeddings

embeddings = from_documents(texts, embedding=FastEmbedEmbeddings(model_name="BAAI/bge-base-en-v1.5", max_length=512))
print(embeddings)
```
NameError: name 'from_documents' is not defined

**Comment by Anush008** [comment]
As per your original message 

> Code to add to chromadb
```
from langchain.vectorstores import Chroma

vectorstore = Chroma.from_documents(documents=texts, embedding=Embedding(), persist_directory="./db2")

```

**Comment by Cyboghostginx** [comment]
```
from langchain.vectorstores import Chroma
from langchain.embeddings.fastembed import FastEmbedEmbeddings

vectorstore = Chroma.from_documents(documents=texts, embedding=FastEmbedEmbeddings(model_name="BAAI/bge-base-en-v1.5", max_length=512), persist_directory="./db2")
print("[+]Done")
```

OperationalError: attempt to write a readonly database

**Comment by Cyboghostginx** [comment]
```
from langchain.vectorstores import Chroma
from langchain.embeddings import OllamaEmbeddings

vectorstore = Chroma.from_documents(documents=texts, embedding=FastEmbedEmbeddings(model_name="BAAI/bge-base-en-v1.5", max_length=512), persist_directory="./db3")
print("[+]Done")
```

I also changed the persisit directory name, now different error.

ValueError: Expected EmbeddingFunction.__call__ to have the following signature: odict_keys(['self', 'input']), got odict_keys(['self', 'args', 'kwargs'])
Please see https://docs.trychroma.com/embeddings for details of the EmbeddingFunction interface.
Please note the recent change to the EmbeddingFunction interface: https://docs.trychroma.com/migration#migration-to-0416---november-7-2023

**Comment by Anush008** [comment]
> ```
> from langchain.vectorstores import Chroma
> from langchain.embeddings.fastembed import FastEmbedEmbeddings
> 
> vectorstore = Chroma.from_documents(documents=texts, embedding=FastEmbedEmbeddings(model_name="BAAI/bge-base-en-v1.5", max_length=512), persist_directory="./db2")
> print("[+]Done")
> ```
> 
> OperationalError: attempt to write a readonly database

Appears to be related to Chroma. Could you try reaching out support?

**Comment by Cyboghostginx** [comment]
Ok thanks, I would

**Comment by Cyboghostginx** [comment]
```
from langchain.vectorstores import Chroma
from langchain.embeddings import OllamaEmbeddings
from chromadb.utils import embedding_functions
from langchain.vectorstores import Qdrant
from langchain.embeddings.fastembed import FastEmbedEmbeddings

embeddings = FastEmbedEmbeddings(model_name="BAAI/bge-base-en-v1.5", max_length=512)

#vectorstore = Chroma.from_documents(documents=texts, embedding=FastEmbedEmbeddings(model_name="BAAI/bge-base-en-v1.5", max_length=512), persist_directory="./db3")
qdrant = Qdrant.from_documents(
    texts,
    embeddings,
    path="./local_qdrant",
    collection_name="db1",
)
print("[+]Done")
```

Qdrant worked perfectly!
